controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/CuratorDb.java

```diff
@@ -55,35 +64,58 @@
 
     /** Create a curator db which also set up a ZooKeeper server (such that this instance is both client and server) */
     @Inject
-    public CuratorDb() {
-        // this.zooKeeperServer = new ZooKeeperServer(createZookeeperServerConfig());
-        // this.curator = new Curator("localhost:2281");
-        //this.zooKeeperServer = null;
-        this.curator = new MockCurator();
-    }
-    
-    private static ZookeeperServerConfig createZookeeperServerConfig() {
+    public CuratorDb(ClusterInfoConfig clusterInfo) {
+        this.zooKeeperServer = new ZooKeeperServer(toZookeeperServerConfig(clusterInfo));
+        this.curator = new Curator(toConnectionSpec(clusterInfo));
+    }
+
+    /** Create a curator db which does not set up a server, using the given Curator instance */
+    protected CuratorDb(Curator curator) {
+        this.zooKeeperServer = null;
+        this.curator = curator;
+    }
+
+    private static ZookeeperServerConfig toZookeeperServerConfig(ClusterInfoConfig clusterInfo) {
         ZookeeperServerConfig.Builder b = new ZookeeperServerConfig.Builder();
         b.zooKeeperConfigFile("conf/zookeeper/controller-zookeeper.cfg");
         b.dataDir("var/controller-zookeeper");
-        b.clientPort(2281);
+        b.clientPort(zooKeeperPort);
         b.myidFile("var/controller-zookeeper/myid");
-        b.myid(0);
-        ZookeeperServerConfig.Server.Builder server = new ZookeeperServerConfig.Server.Builder();
-        server.id(0);
-        server.hostname("localhost");
-        server.quorumPort(2282);
-        server.electionPort(2283);
-        b.server(server);
+        // Ideally we would inject and use the index in ClusterMemberInfoConfig, but the config is always empty for
+        // reasons unknown
+        b.myid(findMyIndex(clusterInfo));
+
+        for (ClusterInfoConfig.Services clusterMember : clusterInfo.services()) {
+            ZookeeperServerConfig.Server.Builder server = new ZookeeperServerConfig.Server.Builder();
+            server.id(clusterMember.index());
+            server.hostname(clusterMember.hostname());
+            server.quorumPort(zooKeeperPort + 1);
+            server.electionPort(zooKeeperPort + 2);
+            b.server(server);
+        }
         return new ZookeeperServerConfig(b);
     }
 
-    /** Create a curator db which does not set uop a server, using the given Curator instance */
-    protected CuratorDb(Curator curator) {
-        //this.zooKeeperServer = null;
-        this.curator = curator;
+    private static Integer findMyIndex(ClusterInfoConfig clusterInfo) {
+        try {
+            String hostname = InetAddress.getLocalHost().getHostName();
```

Comment:
I think we should use HostName.getLocalhost(), that's what we try to everywhere in the config system (so using that would be the same as used when generating the config)

Reply:
Fixed.